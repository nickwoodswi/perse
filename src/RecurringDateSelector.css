.recurrence-selectors{
    width: 98%;
    display: flex;
    justify-content: space-between;
    margin-top: 5px;
}
    .recurrence-selector{
        display: flex;
        align-items: center;
        justify-content: center;
        background-color: yellow;
        width: 70px;
        height: 50px;
    }