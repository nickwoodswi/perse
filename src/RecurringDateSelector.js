import React, { Component } from 'react'
import MonthDayCounts from './MonthDayCounts'
import './RecurringDateSelector.css'

class RecurringDateSelector extends Component{
    render() {
        const months = Object.keys(MonthDayCounts)
        const startDays = MonthDayCounts[this.props.selectedStartMonth]
        const endDays = MonthDayCounts[this.props.selectedEndMonth]
        return(
            <>
            <div className="range-date-divider">RECURRING EVERY:</div>
            <div className="recurrence-selector">
                <select className="recurrence-selector" onChange={e => this.props.setRecurrance('recurrance', e.target.value)}>
                    <option value="0">SELECT RECURRANCE</option>
                    <option value="1">DAY</option>
                    <option value="2">SECOND DAY</option>
                    <option value="3">THIRD DAY</option>
                    <option value="7">WEEK</option>
                </select>
            </div>
            <div className="range-date-divider">STARTING</div>
            <div className="range-date-selector">
                <div className="date-selector">
                    <select 
                        key="start-month-selector" 
                        className="month-selector" 
                        onChange={e => this.props.handleRecurringDate('month_start', e.target.value)}>
                            {months.map((month, idx) => {
                                return(
                                    <option key={idx}>{month}</option>
                                )
                            })}
                    </select>
                    
                    <select 
                        key="start-day-selector" 
                        className="day-selector" 
                        onChange={e => this.props.handleRecurringDate('day_start', e.target.value)}>
                        <>{startDays}</>
                    </select>
                
                    <select 
                        key="start-year-selector" 
                        className="year-selector" 
                        onChange={e => this.props.handleRecurringDate('year_start', e.target.value)}>
                            <option>SELECT YEAR</option>
                            <option>2020</option>
                            <option>2021</option>
                    </select>
                </div>

                <div className="range-date-divider">THRU</div>

                <div className="date-selector">
                    <select 
                        key="end-month-selector" 
                        className="month-selector" 
                        onChange={e => this.props.handleRecurringDate('month_end', e.target.value)}>
                            {months.map((month, idx) => {
                                return(
                                    <option key={idx}>{month}</option>
                                )
                            })}
                    </select>

                    <select 
                        key="end-day-selector" 
                        className="day-selector" 
                        onChange={e => this.props.handleRecurringDate('day_end', e.target.value)}>
                        <>{endDays}</>
                    </select>

                    <select 
                        key="end-year-selector" 
                        className="year-selector" 
                        onChange={e => this.props.handleRecurringDate('year_end', e.target.value)}>
                            <option>SELECT YEAR</option>
                            <option>2020</option>
                            <option>2021</option>
                    </select>
                </div>
            </div>
            </>
        )
    }
}

export default RecurringDateSelector