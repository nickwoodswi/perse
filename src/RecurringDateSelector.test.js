import React from 'react';
import { render } from '@testing-library/react';
import RecurringDateSelector from './RecurringDateSelector';
import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

test('renders RecurringDateSelector', () => {
    render(<MemoryRouter><RecurringDateSelector /></MemoryRouter>);
});

it('RecurringDateSelector renders the UI as expected', () => {
    const tree = renderer
        .create(<MemoryRouter><RecurringDateSelector /></MemoryRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});