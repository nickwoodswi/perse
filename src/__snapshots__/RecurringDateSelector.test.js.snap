// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecurringDateSelector renders the UI as expected 1`] = `
Array [
  <div
    className="range-date-divider"
  >
    RECURRING EVERY:
  </div>,
  <div
    className="recurrence-selectors"
  >
    <div
      className="recurrence-selector"
      onClick={[Function]}
    >
      DAY
    </div>
    <div
      className="recurrence-selector"
      onClick={[Function]}
    >
      2nd DAY
    </div>
    <div
      className="recurrence-selector"
      onClick={[Function]}
    >
      3rd DAY
    </div>
    <div
      className="recurrence-selector"
      onClick={[Function]}
    >
      WEEK
    </div>
  </div>,
  <div
    className="recurrence-indicator"
  >
    DAY
  </div>,
  <div
    className="range-date-divider"
  >
    STARTING
  </div>,
  <div
    className="range-date-selector"
  >
    <div
      className="date-selector"
    >
      <select
        className="month-selector"
        onChange={[Function]}
      >
        <option>
          MONTH
        </option>
        <option>
          January
        </option>
        <option>
          February
        </option>
        <option>
          March
        </option>
        <option>
          April
        </option>
        <option>
          May
        </option>
        <option>
          June
        </option>
        <option>
          July
        </option>
        <option>
          August
        </option>
        <option>
          September
        </option>
        <option>
          October
        </option>
        <option>
          November
        </option>
        <option>
          December
        </option>
      </select>
      <select
        className="day-selector"
        onChange={[Function]}
      />
      <select
        className="year-selector"
        onChange={[Function]}
      >
        <option>
          2020
        </option>
        <option>
          2021
        </option>
      </select>
    </div>
    <div
      className="range-date-divider"
    >
      THRU
    </div>
    <div
      className="date-selector"
    >
      <select
        className="month-selector"
        onChange={[Function]}
      >
        <option>
          MONTH
        </option>
        <option>
          January
        </option>
        <option>
          February
        </option>
        <option>
          March
        </option>
        <option>
          April
        </option>
        <option>
          May
        </option>
        <option>
          June
        </option>
        <option>
          July
        </option>
        <option>
          August
        </option>
        <option>
          September
        </option>
        <option>
          October
        </option>
        <option>
          November
        </option>
        <option>
          December
        </option>
      </select>
      <select
        className="day-selector"
        onChange={[Function]}
      />
      <select
        className="year-selector"
        onChange={[Function]}
      >
        <option>
          2020
        </option>
        <option>
          2021
        </option>
      </select>
    </div>
  </div>,
]
`;
